### PR TITLE
SharedUserCollections: persistence

### PR DESCRIPTION
**Pull request by eronisko**
Migrations, controller, tests

### CODE REVIEW

File: app/SharedUserCollection.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        'name',
+        'author',
+        'description',
+        'items',
+    ];
+
+    protected static function booted()
+    {
+        static::creating(function ($collection) {
+            $collection->public_id = $collection->public_id ?: Str::random(6);
```
Comment (igor-kamil) [inline app/SharedUserCollection.php:25]: no check for unique?

File: resources/views/frontend/shared-user-collections/show.blade.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
+@extends('layouts.master') {{-- TODO figure out layout --}}
+
+@section('title')
+    Zdieľanie kolekcie {{-- TODO i18n --}}
```
Comment (igor-kamil) [inline resources/views/frontend/shared-user-collections/show.blade.php:4]: this is quick to implement....

**Review by igor-kamil — COMMENTED** [review]
overall looks nice. please define translation strings before merge.

File: app/SharedUserCollection.php (hunk 1 above)
Comment (eronisko) [reply app/SharedUserCollection.php:25]: There's an index on the DB side. I've figured the probability of a clash is very low (it's a random string of 6 alphanumeric characters in both cases)

**Review by igor-kamil — APPROVED** [review]
since this is just part of an epic, it's good to go as it is

**Pull request merged**